### ISSUE TITLE
Constructing Str from SubString fails

### ISSUE THREAD

**Issue by bkamins** [JuliaString/Strs.jl#49]
Currently `Str(substring)` falls back to a `convert` method which fails.

Additionally currently recommended design in Julia (AFAIK) is to define constructors and make `convert` methods call them - not the other way around (but probably this is not so crucial).

**Comment by ScottPJones** [comment]
The reason I did it that way, is that there may be many options for `convert`, but I wanted the constructors to just do the simple cases.

**Comment by ScottPJones** [comment]
This should be fixed now, by #73

**Comment by bkamins** [comment]
Still fails (I am using 435ffa234043966ebdbdded35d9):

```
julia> x = "1234"
"1234"

julia> y = SubString(x, 2,3)
"23"

julia> Str(y)
ERROR: MethodError: no method matching _str_encode(::SubString{String}, ::Int64, ::UInt64)
Closest candidates are:
  _str_encode(::T<:Strs.Str, ::Any, ::Any) where T<:Strs.Str at D:\DEV\Julia\Strs.jl\src\encode.jl:88
Stacktrace:
 [1] convert(::Type{Strs.Str}, ::SubString{String}) at D:\DEV\Julia\Strs.jl\src\encode.jl:103
 [2] Strs.Str(::SubString{String}) at D:\DEV\Julia\Strs.jl\src\encode.jl:206
```

**Comment by ScottPJones** [comment]
Fixed with #81 :grinning:
